### PR TITLE
[#151394226] Feature: adding match id in netcode.io packets.

### PR DESCRIPTION
**Pull request by mpope9**
Packets are now created with a match id. The hashtable's key is now a uint64_t instead of an int, for compatibility with TS. All unit tests are passing. Valgrind reports no additional errors. cppcheck is clean.

SEE: https://github.com/skillz/netcode.io/pull/4 for full list of commits.

So I changed the network simulator for the last test.  See here for a better explanation (at the bottom): 
https://github.com/skillz/yojimbo/pull/2

@cchung89 first round
@kkaminski second round

### CODE REVIEW

**Comment by cchung89** [comment]
🎲 LGTM

File: netcode.c [hunk 1]
```diff
@@ -5321,6 +5328,7 @@ static void test_address()
 
 #define TEST_PROTOCOL_ID            0x1122334455667788ULL
 #define TEST_CLIENT_ID              0x1ULL
+#define TEST_MATCH_ID				0x2ULL
```
Comment (cchung89) [inline netcode.c]: Nit, I notice this isn't lining up

Comment (mpope9) [reply netcode.c]: Fixed.

File: netcode.c [hunk 2]
```diff
@@ -5328,7 +5341,7 @@ static void test_address()
 
 #define TEST_PROTOCOL_ID            0x1122334455667788ULL
 #define TEST_CLIENT_ID              0x1ULL
-#define TEST_MATCH_ID				0x2ULL
+#define TEST_MATCH_ID			0x2ULL
```
Comment (cchung89) [inline netcode.c]: It seems the code still doesn't line up, you might want tot use sublime or something, or double-check to make sure it is aligned.

File: netcode.c [hunk 3]
```diff
@@ -41,10 +41,10 @@
 #define NETCODE_SOCKET_IPV6         1
 #define NETCODE_SOCKET_IPV4         2
 
-#define NETCODE_CONNECT_TOKEN_PRIVATE_BYTES 1024
-#define NETCODE_CHALLENGE_TOKEN_BYTES 300
+#define NETCODE_CONNECT_TOKEN_PRIVATE_BYTES ( 1024 + 8 )
+#define NETCODE_CHALLENGE_TOKEN_BYTES ( 300 + 8 )
 #define NETCODE_VERSION_INFO_BYTES 13
-#define NETCODE_USER_DATA_BYTES 256
+#define NETCODE_USER_DATA_BYTES ( 256 + 8 )
```
Comment (cchung89) [inline netcode.c]: What are these +8 for? Is that needed? I also see some other +8 in netcode.c as well.

Comment (mpope9) [reply netcode.c]: I have that to distinguish what netcode.io uses for their default size, then what we are adding.  In this case its only 8 bytes so far.  But in some other spots there is different reasoning...

Comment (mpope9) [reply netcode.c]: Actually this one isn't needed.  Removing it.

**Comment by kkaminski** [comment]
Can you update the title to reflect what's really changed?  That title shows up in annotations and if we ever do a changelist

**Comment by kkaminski** [comment]
🌮 full disclosure - I did not pour through this.  Seems like a lot of test changes but hard to tell since netcode.io is one HUGE file, ugh.  Please update the wiki (game-sync-service) like we talked about and the title before merging.

**Pull request merged**